src/EventStore.Client.Streams/EventStoreClient.Read.cs

```diff
@@ -122,6 +122,96 @@ public partial class EventStoreClient {
 				userCredentials, cancellationToken);
 		}
 
+		/// <summary>
+		///  Represents a fold result
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		public struct FoldResult<T> {
+			/// <summary>
+			/// the position of the last event folded
+			/// </summary>
+			public StreamRevision Revision { get; }
+			/// <summary>
+			/// the fold value
+			/// </summary>
+			public T Value { get; }
+
+			/// <summary>
+			/// build a fold resuult
+			/// </summary>
+			/// <param name="revision"></param>
+			/// <param name="value"></param>
+			public FoldResult(StreamRevision revision, T value) {
+				Revision = revision;
+				Value = value;
+			}
+		}
+
+		/// <summary>
+		/// folds a stream using provided aggregator and seed
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <typeparam name="E"></typeparam>
+		/// <param name="deserialize"></param>
+		/// <param name="aggregator"></param>
+		/// <param name="streamName"></param>
+		/// <param name="revision"></param>
+		/// <param name="seed"></param>
+		/// <param name="configureOperationOptions"></param>
+		/// <param name="resolveLinkTos"></param>
+		/// <param name="userCredentials"></param>
+		/// <param name="cancellationToken"></param>
+		/// <returns></returns>
+		public async ValueTask<FoldResult<T>> FoldStreamAsync<T,E>(
```

Comment:
Can we make this an extension method?

Reply:
I'd like have been able to do this, but it's not possible without access to _client.